test/integration/scheduler/taint_test.go

```diff
@@ -641,9 +643,9 @@ func TestTaintNodeByCondition(t *testing.T) {
 			}
 
 			var pods []*v1.Pod
-			for i, p := range test.pods {
+			for j, p := range test.pods {
 				pod := p.pod.DeepCopy()
-				pod.Name = fmt.Sprintf("%s-%d", pod.Name, i)
+				pod.Name = fmt.Sprintf("%s-%d-%d", pod.Name, i, j)
```

Comment:
If there're still some nodes in the cache, scheduer may schedule pod to it which may trigger flaky result of unschedulable case.

Reply:
@k82cn yeap, that was also my concern. But later on I found there was already some issue/pr fixing the case by checking if the node really exists or not. Generally speaking, it should be all good but don't have extraordinary benefits, so don't worry, I will close this one as your pr has been merged :)